src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -130,15 +130,18 @@ private[health] class HealthCheckActor(
     }
   }
 
-  def ignoreFailures(task: Task, health: Health): Boolean = {
+  def ignoreFailures(instance: Instance, health: Health): Boolean = {
     // Ignore failures during the grace period, until the task becomes green
     // for the first time.  Also ignore failures while the task is staging.
-    task.launched.fold(true) { launched =>
-      health.firstSuccess.isEmpty &&
-        launched.status.startedAt.fold(true) { startedAt =>
-          startedAt + healthCheck.gracePeriod > Timestamp.now()
-        }
-    }
+    Task.from(instance).map { task =>
+      task.launched.fold(true) { launched =>
+        health.firstSuccess.isEmpty &&
+          launched.status.startedAt.fold(true) { startedAt =>
+            startedAt + healthCheck.gracePeriod > Timestamp.now()
+          }
+      }
+    }.getOrElse(false)
+    // TODO(jdef) support health checks for pods
```

Comment:
yes; I left this note not knowing if we'd need any other changes here. Mesos will send task status updates to Marathon w/ health check information and I haven't yet completely grok'd how we process that.